02B-Python-Omop-walkthrough-critical-care/02B-Python-Omop-walkthrough-critical-care.md

```diff
@@ -0,0 +1,2006 @@
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>visit_occurrence_id</th>
+      <th>person_id</th>
+      <th>visit_concept_id</th>
+      <th>visit_type_concept_id</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>0</th>
+      <td>2451</td>
+      <td>2451</td>
+      <td>9203</td>
+      <td>32817</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>2452</td>
+      <td>2452</td>
+      <td>9203</td>
+      <td>32817</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>2453</td>
+      <td>2453</td>
+      <td>9203</td>
+      <td>32817</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>2454</td>
+      <td>2454</td>
+      <td>9203</td>
+      <td>32817</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>2455</td>
+      <td>2455</td>
+      <td>9201</td>
+      <td>32817</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+
+```python
+df['visit_occurrence']['visit_start_year'] =df['visit_occurrence']['visit_start_date'].dt.to_period('Y')
+df['visit_occurrence']['visit_end_year'] =df['visit_occurrence']['visit_end_date'].dt.to_period('Y')
+```
+
+
+```python
+# Join the Visit Occurance dataframe with Concept
+df_visit_concept = pd.merge(df['visit_occurrence'], df_concept, 
+                             left_on='visit_concept_id', 
+                             right_on='concept_id', 
+                             how='left')
+
+# most frequent Visit Occurance concepts
+visit_occurance_concept_cnt = df_visit_concept.groupby(['concept_name', 'visit_start_year'])['person_id'].count().reset_index().sort_values(by='visit_start_year', ascending=False)
+
+
+# rename the columns
+visit_occurance_concept_cnt.rename(columns={'person_id': 'Count', 'concept_name': 'Visit Concept Name','visit_start_year': 'Year'}, inplace=True)
+
+visit_occurance_concept_cnt.head()
+```
+
+
+
+
+<div>
+<style scoped>
+    .dataframe tbody tr th:only-of-type {
+        vertical-align: middle;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>Visit Concept Name</th>
+      <th>Year</th>
+      <th>Count</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>72</th>
+      <td>Inpatient Visit</td>
+      <td>2023</td>
+      <td>3</td>
+    </tr>
+    <tr>
+      <th>71</th>
+      <td>Inpatient Visit</td>
+      <td>2021</td>
+      <td>1</td>
+    </tr>
+    <tr>
+      <th>33</th>
+      <td>Emergency Room Visit</td>
+      <td>2021</td>
+      <td>1</td>
+    </tr>
+    <tr>
+      <th>70</th>
+      <td>Inpatient Visit</td>
+      <td>2019</td>
+      <td>1</td>
+    </tr>
+    <tr>
+      <th>32</th>
+      <td>Emergency Room Visit</td>
+      <td>2019</td>
+      <td>1</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+
+```python
+# Plot using seaborn
+plt.figure(figsize=(20, 6))
```

Comment:
For this plot can we order it be year in ascending order (so RHS should have 2023)?